app/models/journable/historic_active_record_relation.rb

```diff
@@ -253,22 +253,25 @@ def journals_join_statement
   #
   def add_join_on_journables_table_with_created_at_column(relation)
     relation \
-        .joins("INNER JOIN (SELECT id, created_at#{', project_id' if include_projects?(relation)} " \
+        .joins("INNER JOIN (SELECT id, created_at " \
                "FROM \"#{model.table_name}\") AS journables " \
                "ON \"journables\".\"id\" = \"journals\".\"journable_id\"")
   end
 
-  # Join the projects table on journables if :project is in the includes.
+  # Join the projects table on work_package_journals if :project is in the includes.
   # It is needed when projects are filtered by id, and has to be done manually
   # as eager_loading is disabled.
+  # It needs to be `work_package_journals` and not `journables` (the subselect of the work_packages table)
+  # because the journables table will contain the current project and not the project the work package was
+  # in at the time of the journal.
   # Does not work yet for other includes.
   #
   def add_join_projects_on_journables(relation)
```

Comment:
The method name is not right anymore. 
(I probably would not have noticed if the comment had not been updated.)

Reply:
I renamed it to reflect the change. Thanks for spotting.